packages/cloudflare/src/cli/templates/images.ts

```diff
@@ -13,120 +13,463 @@ export type LocalPattern = {
 	search?: string;
 };
 
-let NEXT_IMAGE_REGEXP: RegExp;
+export async function handleImageRequest(
+	requestURL: URL,
+	requestHeaders: Headers,
+	env: CloudflareEnv
+): Promise<Response> {
+	const parseResult = parseImageRequest(requestURL, requestHeaders);
+	if (!parseResult.ok) {
+		return new Response(parseResult.message, {
+			status: 400,
+		});
+	}
 
-/**
- * Fetches an images.
- *
- * Local images (starting with a '/' as fetched using the passed fetcher).
- * Remote images should match the configured remote patterns or a 404 response is returned.
- */
-export async function fetchImage(fetcher: Fetcher | undefined, imageUrl: string, ctx: ExecutionContext) {
-	// https://github.com/vercel/next.js/blob/d76f0b1/packages/next/src/server/image-optimizer.ts#L208
-	if (!imageUrl || imageUrl.length > 3072 || imageUrl.startsWith("//")) {
-		return getUrlErrorResponse();
+	let imageResponse: Response;
+	if (parseResult.url.startsWith("/")) {
+		if (env.ASSETS === undefined) {
+			console.error("env.ASSETS binding is not defined.");
```

Comment:
use the logger rather than console.

Also we should fallback to the former behavior (serve unoptimized images) when the binding is not present)